### ISSUE TITLE
Add CSS compilation

### ISSUE THREAD

**Issue by gigamorph** [ProjectMirador/mirador#1264]
Use [Less](http://lesscss.org) to compile CSS. This will help modularize and maintain style definitions easier. Please assign this issue to me.

**Comment by gigamorph** [comment]
Created a PR at #1270. 

I made a crude attempt to restructure the existing style definitions in Less way, namely multiple .less files with nested styles and parametrization.

I haven't tried to fix any semantic errors or duplicates/redundancies this time, but hopefully the new structure will make it easier to find and fix those.

One thing I did is parametrizing color definitions. All colors are ultimately maintained as constants in `theme/colors.less`. You can see the definitions there are rather messy, revealing need for re-theming the whole set of styles and maybe reduce the number of colors used in a consistent way. Then we can do a secondary parametrization, such as 
```
@off-white: #fafafa;
@editor-text-color: @off-white;
.annotation-editor textarea {
  color: @editor-text-color;
}
```

Other aspects can be parametrized and unified in the same way, e.g., fonts, paddings, etc., after which we could even create different themes of styles for the user to choose from.

Note: all styles are now defined under the selector `.mirador-container`. Which means the root element should have that class for the styles to be applied. For example, see the change in `index.html`:
```
<div id="viewer" class="mirador-container"></div>
```

**Comment by aeschylus** [comment]
Hey @gigamorph. We're working on reviewing and merging this PR, but we're concerned that this might be a breaking change (requiring the labelling of the parent div). One solution is to generate (as part of the viewer template) a containing element with the correct class(es).

**Comment by rsinghal** [comment]
Could you also check to see if the element given to Mirador has the class?  If not, add it.

**Comment by gigamorph** [comment]
@aeschylus @rsinghal Can I just add the class to the viewer (`.mirador-viewer`)'s immediate parent, as @csbailey5t suggested in #1270?

**Comment by rsinghal** [comment]
That's what I was suggesting.  I think it's the cleanest solution.

**Comment by aeschylus** [comment]
Yeah, I don't see why not since we aren't changing any properties that might interfere with the parent's behaviour on the host page.

**Comment by gigamorph** [comment]
PR is updated with the change now.

**Comment by csbailey5t** [comment]
Closed in #1270
